ios/App/App/Sample Plugin/R2Plugin.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+import Capacitor
+import Combine
+import R2Shared
+import R2Streamer
+import UIKit
+
+@objc(R2Plugin)
+public class R2Plugin: CAPPlugin {
+    
+    private var pluginReceiver: CAPPluginCall?
+    
+    private lazy var module: SampleModule = {
+        let module = SampleModule()
+        module.delegate = self
+        return module
+    }()
+    
+    // named openBook temporarily to work with the existing setup
+    // perform the first time setups
+    @objc func openBook(_ call: CAPPluginCall) {
+        call.keepAlive = true
+        pluginReceiver = call
+
+        let example: [SampleModel] = [
+            SampleModel(id: "1", title: "example 1"),
+            SampleModel(id: "2", title: "example 2"),
+            SampleModel(id: "3", title: "example 3")
+        ]
+
+        DispatchQueue.main.async {
+            self.bridge?.viewController?.present(self.module.viewController, animated: true, completion: nil)
+            self.module.receiveModels(models: example)
+        }
+    }
+    
+    @objc func update(_ call: CAPPluginCall) {
+        guard let array = call.getArray("samples") else {
+            call.reject("Could not parse plugin call")
+            return
+        }
+        
+        let models: [SampleModel] = array.compactMap {
+            guard let element = $0 as? JSObject,
+                  let id = element["id"] as? String,
+                  let title = element["title"] as? String else { return nil }
+            return SampleModel(id: id, title: title)
+        }
+        
+        DispatchQueue.main.async {
+            self.module.receiveModels(models: models)
+        }
+                
+        call.resolve()
+    }
+    
+    @objc func displayError(_ call: CAPPluginCall) {
+        let message = call.getString("message")
+        
+        let alert = UIAlertController(title: nil, message: message, preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
+        
+        self.bridge?.viewController?.present(alert, animated: true, completion: nil)
+        
+        call.resolve()
+    }
+}
+
+extension R2Plugin: SampleModuleDelegate {
+
+    func createSampleModel(sampleModel: SampleModel) {
+        pluginReceiver?.resolve([
```

Comment:
As far as I can tell, it would be safe to assume that the hybrid catalog will always be the one to initiate the contact with the native module -- meaning the setup call can be expected to occur before any updates are made (also the UI that would trigger these operations would not be accessible without a plugin to first setup / present it)